scripts/globals/items/vial_of_refresh_musk.lua

```diff
@@ -1,19 +1,32 @@
 -----------------------------------------
 -- ID: 18241
--- Item: Refresh Musk
+-- Item: Vial of Refresh Musk
 -- Item Effect: 60 seconds
+-- Duration: 30 Seconds
 -----------------------------------------
 require("scripts/globals/status")
-require("scripts/globals/msg")
 
 function onItemCheck(target)
+    local effect = target:getStatusEffect(dsp.effect.ENCHANTMENT)
+    if effect ~= nil and effect:getSubType() == 18241 then
+        target:delStatusEffect(dsp.effect.ENCHANTMENT)
+    end
     return 0
 end
 
 function onItemUse(target)
-    if (not target:hasStatusEffect(dsp.effect.REFRESH)) then
-        target:addStatusEffect(dsp.effect.REFRESH,1,3,60)
+    if target:hasStatusEffect(dsp.effect.ENCHANTMENT) == true then
+        target:delStatusEffect(dsp.effect.ENCHANTMENT)
+        target:addStatusEffect(dsp.effect.ENCHANTMENT,0,0,30,18241)
     else
-        target:messageBasic(dsp.msg.basic.NO_EFFECT)
+        target:addStatusEffect(dsp.effect.ENCHANTMENT,0,0,30,18241)
```

Comment:
erm, both condition branches add the status effect? just verifying this is what you intend here, and what retail does.

Reply:
Just based on this comment i found on ffxiah.com. its a very rare item as the rest of them. The only one i have personal experience is with blm item. 
https://www.ffxiah.com/item/18241/refresh-musk